remarkapy/api.py

```diff
@@ -206,16 +206,157 @@ def list_documents(self) -> list[Entry]:
             RemarkableAPIError: If the request fails.
 
         """
-        url = URLS.LIST_DOCS
+
+        self._refresh_token()
+
+        url = URLS.LIST_ROOT
         headers = {
             "Authorization": f"Bearer {self._config.usertoken}",
             "user-agent": "remarkapy",
         }
-        
+
         response = self._get(url, headers=headers)
         if not response.status_code == 200:
             raise RemarkableAPIError(
                 f"Request failed with status code {response.status_code} when listing documents: {response.text}"
             )
-        results = response.json()
-        return parse_entries(results, fail_method="warn")
+
+        # NOTE we could store this hash somewhere as it probably changes only
+        # when a folder/file gets created/modified (if we decide to cache some data)
+        root_hash = response.json()['hash']
+
+        # List doc hashes on root folder
+        url = URLS.GET_FILE + root_hash
+        response = self._get(url, headers=headers)
+        obj_list = response.text.splitlines()
+
+        root = {}
+
+        # Add a file or a folder to a specific parent_hash or to root
+        def add_item_to_folder(folders, folder_hash: str, parent_hash: str = "", visible_name: str = "", metadata: dict = None, files: list = None):
```

Comment:
Why the name `folders` for this first argument? From what I can tell this only takes a single item?

Reply:
True that. This method was originally meant to match folders to their parents hence the first argument, which sounds out of place now.
 
I feel like we'll get back to this method soon...It does work but it could be simplified.

or we could do something similar to what was done here:
https://github.com/subutux/rmapy/blob/master/rmapy/collections.py